hibernate-core/src/main/java/org/hibernate/sql/exec/internal/JdbcParameterBindingImpl.java

```diff
@@ -17,7 +19,9 @@ public class JdbcParameterBindingImpl implements JdbcParameterBinding {
 	private final Object bindValue;
 
 	public JdbcParameterBindingImpl(JdbcMapping jdbcMapping, Object bindValue) {
-		assert bindValue == null || jdbcMapping.getJdbcJavaType().isInstance( bindValue );
+		assert bindValue == null || jdbcMapping == null || jdbcMapping.getJdbcJavaType().isInstance( bindValue )
```

Comment:
Most uses of `JdbcParameterBinding#getBindType()` assume `jdbcMapping` to be not null. The only place that seems to handle null, appears to be a fragment of the past AFAICT.

Reply:
Its not like I added this one Christian.  And there are in fact (one) uses that set this to null.  If you say that is an "invalid" path now, great - feel free to remove it.  But as it is, it is possible that the `jdbcMapping` is null.  For sure the previous code assumed it was not null as the assertion would fail otherwise, which it never does **in our testsuite**.  But we have been harshly reminded in developing 6.x that our testsuite is far from exhaustive.